include_dir/src/dir.rs

```diff
@@ -1,75 +1,51 @@
 use crate::file::File;
 use std::path::Path;
+use std::ffi::OsStr;
+
+use crate::DirEntry;
 
 /// A directory entry.
 #[derive(Debug, Copy, Clone, PartialEq)]
 pub struct Dir<'a> {
     #[doc(hidden)]
     pub path: &'a str,
     #[doc(hidden)]
-    pub files: &'a [File<'a>],
+    pub file_name: Option<&'a str>,
```

Comment:
`file_name` (and methods) can actually be removed completely. I initially included these as fields because I _thought_ I saw an allocation somewhere within `Path::components()` but I (1) can't find it and (2) that's a premature optimization anyways. I will remove this field and the methods.